packages/main/src/Slider.js

```diff
@@ -0,0 +1,457 @@
+	}
+
+	// Update Slider UI after entering the DOM
+	// TODO Refactor this
+	_initialUISync() {
+		const max = this.max;
+		const min = this.min;
+		const startValue = this.value;
+		const endValue = this.endValue;
+		const sliderDomRect = this.getBoundingClientRect();
+
+		if (!endValue) {
+			const handleDom = this.shadowRoot.querySelector(".ui5-slider-handle");
+			const percentageComplete = (startValue - min) / (max - min);
+			const translatePx = percentageComplete * sliderDomRect.width;
+
+			// Update the position of the handle depending on the value
+			// Center the Slider Handle position under the cursor/pointer
+			handleDom.style.setProperty("left", `${translatePx}px`);
+			this.shadowRoot.querySelector(".ui5-slider-progress").style.setProperty("transform", `scaleX(${percentageComplete})`);
+		} else {
+			const percentageCompleteSecondHandle = (endValue - min) / (max - min);
+			const percentageComplete = (startValue - min) / (max - min);
+			const translatePx = percentageComplete * sliderDomRect.width;
+			const translatePxSecondHandle = percentageCompleteSecondHandle * sliderDomRect.width;
+			const rangeProgress = ((endValue - startValue) - min) / (max - min);
+			this.shadowRoot.querySelector(".ui5-slider-handle").style.setProperty("left", `${translatePx}px`);
+			this.shadowRoot.querySelector(".ui5-slider-end-handle").style.setProperty("left", `${translatePxSecondHandle}px`);
+			this.shadowRoot.querySelector(".ui5-slider-progress").style.setProperty("left", `${translatePx}px`);
+			this.shadowRoot.querySelector(".ui5-slider-progress").style.setProperty("transform", `scaleX(${rangeProgress})`);
+		}
+	}
+
+	// TODO Allow tickmark labeling?
+	/**
+	 * Calculates and draws the tickmarks with a CSS gradient style
+	 */
+	drawDefaultTickmarks(step, max, min) {
+		const stepStr = String(step);
+		const maxStr = String(max);
+		const minStr = String(min);
+
+		// Calculate how many tickmarks have to be drawn (max - min / stepValue)
+		const tickmarksAmount = `${maxStr - minStr} / ${stepStr}`;
+
+		// TODO* Make it themable
+		const tickmarkWidth = "1px";
+
+		// Transparent CSS gradient background
+		const tickmarksGradientBase = `linear-gradient(to right, currentColor ${tickmarkWidth}, transparent 0)`;
+
+		// Draw the tickmarks as a patern over the gradient background
+		const tickmarksGradientdPattern = `0 center / calc((100% - ${tickmarkWidth}) / (${tickmarksAmount})) 100% repeat-x`;
+
+		// Combine to get the complete CSS background gradient property value
+		const tickmarksBackground = `${tickmarksGradientBase + tickmarksGradientdPattern}`;
+
+		// Apply the style to the container
+		this.shadowRoot.querySelector(".ui5-slider-tickmarks").style.setProperty("background", tickmarksBackground);
+	}
+
+	setStep(step) {
+		if (typeof step !== "number" || step < 0) {
+			step = 1;
+		}
+		this.step = step;
+	}
+
+	onEnterDOM() {
+		this._initialUISync() 
+	}
+
+	onBeforeRendering() {
+		if (this.step !== 1) {
```

Comment:
code styling: move the lifecycle methods at the beginning, right after the constructor, as we have them there usually.